### DISCUSSION TITLE
KyselyStore - Expose underlying client

### DISCUSSION THREAD

**Discussion by LoneRifle** [telegraf/telegraf#1937] [Ideas]
This idea proposes exposing the Kysely client that KyselyStore and its dependents (Postgres, MySQL, SQLite) rely on.

It is quite common when constructing Telegram bots for a developer to interact with a database to hold and retrieve state and information that the bot needs. It would hence be helpful if the Kysely client that is used by the relevant Telegraf SessionStores is accessible via a readonly property, so that it could be reused by the developer for their own needs.

Similar changes could be made to the other SessionStores.

**Comment by MKRhere** [comment]
This should ideally be in https://github.com/telegraf/session. Anyhow, All adapters support taking an existing client. In which case, you could create a Kysely instance first, and reuse the underlying driver reference from Kysely with Telegraf Sessions. Would that work for you? Depending on the Sessions package to initialise your Kysely instance seems like a sketchy dependency tree.

**Reply by LoneRifle** [reply]
Thanks for the reply! I checked KyselyStore and notice that it currently does not accept a Kysely instance as part of the ctor options. In other words, KyselyStore currently accepts NewClientOpts, but not ExistingClientOpts. I don't mind filing a PR to add the needed interface to allow this, subject to availability. 

That said, I'm trying to avoid explicitly referencing the `kysely` package in the [template project](https://github.com/opengovsg/telegovsg)'s `package.json` that we have, as my target audience is people who are relatively new to software development. I would hence want as few declared dependencies in `package.json` as possible, to try to avoid (in an admittedly very contrived manner) our target audience making mistakes where we can.

**Reply by MKRhere** [reply]
I'll accept a PR adding ExistingClientOpts to KyselyStore. I hadn't done that because I did not expect KyselyStore to be used directly and not through one of its applications (MySQL, Pg)
